### PR TITLE
feat: move vega charts to API

### LINKED ISSUE #123: API / component proposal for Charts
### What
- We start with histogram.
- We bear in mind scattered plots.
- We use Vega-lite.

### Mockups
- Image

### Part of
- #62

### PR DESCRIPTION
**Pull request by Marc-AntoineA**
### What
This PR implements moves vega charts serverside and generalizes it a little (we just have to add `<searchalicious-chart name="..."/>` clientside).

Resolves #123

### CODE REVIEW

File: app/validations.py [hunk 1]
```diff
@@ -40,3 +40,15 @@ def check_all_facets_fields_are_agg(
...
+def check_all_charts_are_valid(charts: list[str] | None) -> list[str]:
+    """Check all charts are valid,
+    that is, are in the hardcoded list"""
+    errors: list[str] = []
+    if charts is None:
+        return errors
+    for field_name in charts:
+        if field_name not in ["nutriscore_grade", "nova_group", "ecoscore_grade"]:
+            errors.append(f"Unknown field name in charts: {field_name}")
+    return errors
```
Comment (alexgarel) [inline app/validations.py:54]: It should be the same as facets: a field with an aggregate: true property.

File: frontend/src/mixins/search-ctl.ts [hunk 2]
```diff
@@ -292,11 +292,13 @@ export const SearchaliciousSearchMixin = <T extends Constructor<LitElement>>(
         page?: string;
         index?: string;
         facets?: string;
+        charts?: string;
       } = {
         q: queryParts.join(' '),
         langs: this.langs,
         page_size: this.pageSize.toString(),
         index: this.index,
+        charts: ['nutriscore_grade', 'nova_group', 'ecoscore_grade'].join(',')
```
Comment (alexgarel) [inline frontend/src/mixins/search-ctl.ts:301]: It should be a property.

File: app/query.py [hunk 3]
```diff
@@ -291,6 +291,9 @@ def create_aggregation_clauses(
             if field.bucket_agg:
                 # TODO - aggregation might depend on agg type or field type
                 clauses[field.name] = A("terms", field=field.name)
+    clauses['nutriscore_grade'] = A("terms", field='nutriscore_grade')
+    clauses['nova_group'] = A("terms", field='nova_group')
+    # TODO: add  A("terms", field=field.name) for field.name in charts
```
Comment (alexgarel) [inline app/query.py:296]: It should be as for facets, so you should just merge charts and facets when you call this method.

File: frontend/src/mixins/search-ctl.ts (hunk 2 above)
Comment (Marc-AntoineA) [reply frontend/src/mixins/search-ctl.ts]: Same as `facets`, charts is built from `<searchalicious-chart name=.../>` nodes.

File: app/query.py (hunk 3 above)
Comment (Marc-AntoineA) [reply app/query.py]: Done.

File: app/charts.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                            "text": {"signal": "tooltip.amount"},
+                            "fillOpacity": [
+                                {"test": "datum === tooltip", "value": 0},
+                                {"value": 1},
+                            ],
+                        },
+                    },
+                },
+            ],
+        }
```
[the remark this answers is not in the document]
Comment (Marc-AntoineA) [reply app/charts.py:120]: After discussion, we will keep it like this right now. Might be re-discussed with Alex.

File: app/validations.py [hunk 5]
```diff
@@ -22,19 +22,21 @@ def check_index_id_is_defined(index_id: str | None, config: Config) -> None:
...
     errors: list[str] = []
-    if facets is None:
+    if values is None:
         return errors
     global_config = cast(Config, CONFIG)
     index_id, index_config = global_config.get_index_config(index_id)
     if index_config is None:
         raise ValueError(f"Cannot get index config for index_id {index_id}")
-    for field_name in facets:
+    print(index_config)
```
Comment (Marc-AntoineA) [inline app/validations.py:38]: fix me

File: frontend/src/search-chart.ts [hunk 6]
```diff
@@ -14,13 +14,8 @@ export class SearchaliciousChart extends SearchaliciousResultCtlMixin(
...
-  // (eg. key)
   @property()
-  key?: string;
-
-  @property()
-  label?: string;
+  name = '';
 
   @property({type: Array})
   categories: Array<string> = [];
```
Comment (Marc-AntoineA) [inline frontend/src/search-chart.ts:21]: remove me

**Pull request merged**